crates/ruff_python_formatter/src/comments/format.rs

```diff
@@ -32,27 +32,30 @@ pub(crate) enum FormatLeadingComments<'a> {
 
 impl Format<PyFormatContext<'_>> for FormatLeadingComments<'_> {
     fn fmt(&self, f: &mut PyFormatter) -> FormatResult<()> {
-        let comments = f.context().comments().clone();
-
-        let leading_comments = match self {
-            FormatLeadingComments::Node(node) => comments.leading(*node),
-            FormatLeadingComments::Comments(comments) => comments,
-        };
+        fn write_leading_comments(
```

Comment:
The entire change is that I introduced a `write` function that is called inside of the match arms. The formatting logic itself remains unchanged.